### PR TITLE
[r] Add `drop_levels` to `SOMAExperimentAxisQuery` -> ecosystem outgestors

### LINKED ISSUE #2765: [Feature request] `ExperimentAxisQuery.to_anndata()` (and R analogs) should drop unused categories on axis data frames
**Is your feature request related to a problem? Please describe.**
Some analytical pipelines, specially those that relate to visualizations rely on the categories of `pandas.Categorical`. In the case of large SOMAExperiments, many times a query will result on unused categories for potentially many columns of `obs` or `var`, thus the user needs to **always** iterate on all columns and perform a `cat.remove_unused_categories()` operation.

See for example this reproducible example

```python
import cellxgene_census
import scanpy as sc
census = cellxgene_census.open_soma(census_version="2024-05-20")

human = census["census_data"]["homo_sapiens"]
query = human.axis_query(
    measurement_name = "RNA",
    obs_query = tiledbsoma.AxisQuery(
        value_filter = "tissue == 'tongue' and is_primary_data == True"
    )
)

adata = query.to_anndata(column_names={"obs": ["tissue"]}, X_name = "normalized")
sc.pp.pca(adata)
sc.pp.neighbors(adata)
sc.tl.umap(adata)
sc.pl.umap(adata, color="tissue")
```

Only one `"tissue"` was selected but **all** hundreds of tissues are drawn in the umap

image

**Describe the solution you'd like**
`ExperimentAxisQuery.to_anndata()` returns an anndata with unused categories already removed in the axis data frames

### PR DESCRIPTION
**Pull request by mojaveazure**
R analog of #2811 and single-cell-data/SOMA#204; add a `drop_levels` paramter to the ecosystem outgestors to drop unused factor levels from resulting data frames

Modified SOMA methods:
 - `SOMAExperimentAxisQuery$to_seurat()`: add `drop_levels` to drop drop unused levels from `obs` and `var` data frames
 - `SOMAExperimentAxisQuery$to_seurat_assay()`: add `drop_levels` to drop unused levels from `var` data frame
 - `SOMAExperimentAxisQuery$to_single_cell_experiment()`: add `drop_levels` to drop unused levels from `obs` and `var` data frames

Also shifts `SOMAExperimentAxisQuery$to_seurat()` and `SOMAExperimentAxisQuery$to_seurat_assay()` to use `SOMAExperimentAxisQuery$private$.load_df()` for loading `obs` and `var`; removing standalone code and increase sharing with the SCE outgestor

resolves #2765

[SC-51945](https://app.shortcut.com/tiledb-inc/story/51945)

### CODE REVIEW

File: apis/r/R/SOMAExperimentAxisQuery.R [hunk 1]
```diff
@@ -515,6 +515,7 @@ SOMAExperimentAxisQuery <- R6::R6Class(
     #' @param obsm_layers \Sexpr[results=rd]{tiledbsoma:::rd_outgest_mlayers()}
     #' @param varm_layers \Sexpr[results=rd]{tiledbsoma:::rd_outgest_mlayers(axis = 'varm')}
     #' @param obsp_layers \Sexpr[results=rd]{tiledbsoma:::rd_outgest_players()}
+    #' @param drop_levels Drop unused levels from \code{obs} and \code{var} factor columns
```
Comment (eddelbuettel) [inline apis/r/R/SOMAExperimentAxisQuery.R:518]: I do not use these patterns when I write docs so pardon the naive question but why do all other params here follow a pattern and this one doesn't?

**Review by eddelbuettel — APPROVED** [review]
Looks good per the added tests.

One mostly curiously question about Rd writing style inline

**Review by pablo-gar — APPROVED** [review]

File: apis/r/R/SOMAExperimentAxisQuery.R (hunk 1 above)
Comment (mojaveazure) [reply apis/r/R/SOMAExperimentAxisQuery.R:518]: The previous ones used `\Sexpr` for templated docs that are similar but with minor variations between different methods as `@template` does not work with R6; however, the use of `\Sexpr` is the cause of the "installing package to process help pages" step when building tiledbsoma-r from source. As this step has been a source of complaint, I've resorted to copy-pasting docs and will eventually switch out all `\Sexpr` to copy-pasted docs to make building tiledbsoma-r easier

Comment (eddelbuettel) [reply apis/r/R/SOMAExperimentAxisQuery.R:518]: I am 100% on board with the simplification and robustification.

**Pull request merged**